src/main/java/com/zendesk/maxwell/producer/partitioners/AbstractMaxwellPartitioner.java

```diff
@@ -5,30 +5,36 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.UUID;
 
 public abstract class AbstractMaxwellPartitioner {
-	HashStringProvider provider;
 	List<String> partitionColumns = new ArrayList<String>();
+	private final PartitionBy partitionBy, partitionByFallback;
 
-	String partitionKeyFallback;
+	private PartitionBy partitionByForString(String key) {
+		if ( key == null )
+			return null;
 
-	public AbstractMaxwellPartitioner(String partitionKey, String csvPartitionColumns, String partitionKeyFallback) {
-		switch (partitionKey) {
-			case "table": this.provider = new HashStringTable();
-				break;
-			case "primary_key": this.provider = new HashStringPrimaryKey();
-				break;
-			case "column": this.provider = new HashStringColumn();
-				break;
+		switch(key) {
+			case "table":
+				return PartitionBy.TABLE;
 			case "database":
+				return PartitionBy.DATABASE;
+			case "primary_key":
+				return PartitionBy.PRIMARY_KEY;
+			case "column":
+				return PartitionBy.COLUMN;
 			default:
-				this.provider = new HashStringDatabase();
-				break;
+				return null;
```

Comment:
The default was by database before?

Reply:
the default for invalid strings was database, yeah.  I think if you pass an unknown string this should throw a runtime error, but if you pass `null` you get `PartitionBy.DATABASE`.  Since this is also used to determine the fallback partition-by I think that'll cover both comments.